src/server/debug/server/server.go

```diff
@@ -101,9 +102,53 @@ func NewDebugServer(env Env) debug.DebugServer {
 		logLevel:      log.LogLevel,
 		grpcLevel:     log.GRPCLevel,
 	}
+	ctx := pctx.Background("debug server")
+	go s.logDBState(ctx)
 	return s
 }
 
+func (s *debugServer) logDBState(ctx context.Context) {
+	ticker := time.NewTicker(3 * time.Second)
+	queries := map[string]string{
+		"idle queries": `
+                  SELECT current_timestamp - query_start AS runtime,
+                         datname,
+                         usename,
+                         client_addr,
+                         backend_xid,
+                         query
+		  FROM pg_stat_activity WHERE state = 'idle' AND current_timestamp - query_start > interval '10 seconds' ORDER BY runtime DESC;`,
```

Comment:
On my machine, I see state = 'idle in transaction' instead of just idle.  Maybe change = to LIKE?